iceoryx_hoofs/CMakeLists.txt

```diff
@@ -48,8 +48,13 @@ iox_add_library(
     PRIVATE_LIBS                ${ICEORYX_SANITIZER_FLAGS}
     PRIVATE_LIBS_LINUX          acl atomic ${CODE_COVERAGE_LIBS}
     BUILD_INTERFACE             ${PROJECT_SOURCE_DIR}/include
+                                ${PROJECT_SOURCE_DIR}/legacy/include
+                                ${PROJECT_SOURCE_DIR}/memory/include
                                 ${CMAKE_BINARY_DIR}/generated/iceoryx_hoofs/include
     INSTALL_INTERFACE           include/${PREFIX}
+    EXPORT_INCLUDE_DIRS         include/
+                                legacy/include/
+                                memory/include/
```

Comment:
@elBoberido I duplicated it because Bazel does duplicate it as well and it felt better to be in control of both build an include directories.

Reply:
Happy to refactor if you know of a nicer solution.

Reply:
It just feels a bit superfluous but on the other hand you are right, it gives us a bit more control. @FerdinandSpitzschnueffler what's your opinion on this?

Reply:
@mossmaurice could you check if something like `EXPORT_INCLUDE_DIRS    ${BUILD_INTERFACE}` would work? With this we could eat the cake and have it too :)

Reply:
@elBoberido I did:

```
    if ( NOT IOX_NO_EXPORT )
        setup_install_directories_and_export_package(
            TARGETS ${IOX_TARGET} ${IOX_ADDITIONAL_EXPORT_TARGETS}
            INCLUDE_DIRECTORIES ${IOX_BUILD_INTERFACE}
        )
    endif()
```

An that broke the out-of-tree build. My feeling is still it should not be an implicit thing, but explicit defined in our custom macro.

Reply:
@mossmaurice you got me wrong. I didn't meant to make it implicit but to explicitly reuse the header from the `BUILD_INTERFACE`. I think they are locally stored in `IOX_BUILD_INTERFACE`. In most cases one would then just have a `EXPORT_INCLUDE_DIRS ${IOX_BUILD_INTERFACE}` in the `iox_add_library` call. This would remove duplication.

Reply:
Well, that example is a workaround for the bug you fixed :)
Further down you will notice that `${CMAKE_BINARY_DIR}/generated/iceoryx_hoofs/include` will also be installed. This workaround could then be removed

Reply:
Okay, i tried it and it seems it does not work. Maybe we can improve it in the future but it is okay as is for now.